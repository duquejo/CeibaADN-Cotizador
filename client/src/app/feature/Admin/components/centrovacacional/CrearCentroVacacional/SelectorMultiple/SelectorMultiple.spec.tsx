import * as React from 'react';
import { SinonStub, stub } from 'sinon';
import { SelectorMultiple } from '.';
import { shallow, ShallowWrapper } from 'enzyme';

describe('SelectorMultiple test', () => {
    
  let componentWrapper: ShallowWrapper;
  let componentProps: React.ComponentProps<typeof SelectorMultiple> & {
    onSubmit: SinonStub;
    handlerSelection: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
        handlerSeleccion: stub(),
        options: [
            {
                id: 5,
                nombre: 'Categoría de usuarios de prueba',
                descripcion: 'Descripción de categoría de usuarios',
                valorAlta: 64000,
                valorBaja: 30000
            },
            {
                id: 12,
                nombre: 'Otra categoría de prueba',
                valorAlta: 32000,
                valorBaja: 12000
            }
      ],
      containerClass: 'clase_prueba',
      initialValues: [{
        label: 'Categoría de usuarios de prueba',
        value: 5
      }],
      selectText: 'Prueba - Seleccionar todos'
    },

    jest.clearAllMocks();
    componentWrapper = shallow( <SelectorMultiple {...componentProps} /> );
  });

  afterEach(() => {
    componentWrapper.unmount();
  });

  test('Debería renderizar <SelectorMultiple /> satisfactoriamente', () => {
    expect( componentWrapper ).toMatchSnapshot();
  });

  test('Debería verificar si el contenedor posee la clase establecida', () => {

    const container = componentWrapper.find('MultiSelect');
    expect(container.hasClass(componentProps.containerClass)).toBeTruthy();
  });  
});
