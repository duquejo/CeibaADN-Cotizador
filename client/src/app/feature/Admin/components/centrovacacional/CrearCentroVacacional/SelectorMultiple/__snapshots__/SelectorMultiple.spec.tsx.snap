// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorMultiple test Debería renderizar <SelectorMultiple /> satisfactoriamente 1`] = `
<Fragment>
  <MultiSelect
    className="selector__container clase_prueba"
    labelledBy="Seleccionar"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Categoría de usuarios de prueba",
          "value": 5,
        },
        Object {
          "label": "Otra categoría de prueba",
          "value": 12,
        },
      ]
    }
    overrideStrings={
      Object {
        "allItemsAreSelected": "Todos los elementos están seleccionados",
        "clearSearch": "Limpiar búsqueda",
        "clearSelected": "Limpiar selección",
        "create": "Prueba - Seleccionar todos",
        "noOptions": "No hay opciones",
        "search": "Buscar",
        "selectAll": "Seleccionar todos",
        "selectAllFiltered": "Seleccionar todos",
        "selectSomeItems": "Prueba - Seleccionar todos",
      }
    }
    value={Array []}
  />
</Fragment>
`;
